src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -345,29 +358,62 @@ public void Read(JsonTokenType tokenType, ref ReadStack state, ref Utf8JsonReade
                 // Forward the setter to the value-based JsonPropertyInfo.
                 propertyInfo.ReadEnumerable(tokenType, ref state, ref reader);
             }
+            // For performance on release build, don't verify converter correctness for internal converters.
+            else if (HasInternalConverter)
+            {
+#if DEBUG
+                JsonTokenType originalTokenType = reader.TokenType;
+                int originalDepth = reader.CurrentDepth;
+                long originalBytesConsumed = reader.BytesConsumed;
+#endif
+
+                OnRead(ref state, ref reader);
+
+#if DEBUG
```

Comment:
nit: The `HasInternalConverter` code paths is introducing code duplication. Extract into a helper/avoid the duplication.

Reply:
It is not possible to avoid code duplication here since the DEBUG statements are only for HasInternalConverter==true. Even if it was possible to avoid code duplication this avoids an extra function call which may not be inlined (code is on a hot path).